lib/tbd/ua.rb

```diff
@@ -593,77 +593,46 @@ def ua_summary(date = Time.now, argh = {})
-      end
-
-      if surface.key?(:doors)
-        surface[:doors].values.each do |door|
-          next unless door.key?(:gross)
-          next unless door[:gross] > TOL
-          next unless door.key?(:u)
-          next unless door[:u] > TOL
-          areas[:walls][:subs ] += door[:gross]              if type == :wall
-          areas[:roofs][:subs ] += door[:gross]              if type == :ceiling
-          areas[:floors][:subs] += door[:gross]              if type == :floor
-           bloc[:pro][:doors  ] += door[:gross] * door[:u]
-
-           ok = door.key?(:ref)
-           bloc[:ref][:doors  ] += door[:gross] * door[:ref]               if ok
-           bloc[:ref][:doors  ] += door[:gross] * door[:u ]            unless ok
-        end
-      end
-
-      if surface.key?(:windows)
-        surface[:windows].values.each do |window|
-          next unless window.key?(:gross)
-          next unless window[:gross] > TOL
-          next unless window.key?(:u)
-          next unless window[:u] > TOL
-          areas[:walls][:subs ]  += window[:gross]           if type == :wall
-          areas[:roofs][:subs ]  += window[:gross]           if type == :ceiling
-          areas[:floors][:subs]  += window[:gross]           if type == :floor
-           bloc[:pro][:windows]  += window[:gross] * window[:u]
-
-          ok = window.key?(:ref)
-          bloc[:ref][:windows ] += window[:gross] * window[:ref]           if ok
-          bloc[:ref][:windows ] += window[:gross] * window[:u  ]       unless ok
-        end
+        bloc[:pro][:floors ] += surface[:net] * surface[:u  ]
+        bloc[:ref][:floors ] += surface[:net] * surface[:ref]       if reference
+        bloc[:ref][:floors ] += surface[:net] * surface[:u  ]   unless reference
       end
 
-      if surface.key?(:skylights)
-        surface[:skylights].values.each do |sky|
-          next unless sky.key?(:gross)
-          next unless sky[:gross] > TOL
-          next unless sky.key?(:u)
-          next unless sky[:u] > TOL
-          areas[:walls][:subs  ] += sky[:gross]              if type == :wall
-          areas[:roofs][:subs  ] += sky[:gross]              if type == :ceiling
-          areas[:floors][:subs ] += sky[:gross]              if type == :floor
-          bloc[:pro][:skylights] += sky[:gross] * sky[:u]
-
-          ok = sky.key?(:ref)
-          bloc[:ref][:skylights] += sky[:gross] * sky[:ref]                if ok
-          bloc[:ref][:skylights] += sky[:gross] * sky[:u  ]            unless ok
+      [:doors, :windows, :skylights].each do |subs|
+        next unless surface.key?(subs)
+
+        surface[subs].values.each do |sub|
+          next unless sub.key?(:gross)
+          next unless sub.key?(:u    )
+          next unless sub[:gross] > TOL
+          next unless sub[:u    ] > TOL
+
+          gross  = sub[:gross]
+          gross *= sub[:mult ]                               if sub.key?(:mult)
```

Comment:
Keeping (better) track of subsurface multipliers. More tests (and potentially fixes) on the way.